src/cmodel.c

```diff
@@ -1272,24 +1272,68 @@ void CM_InvalidateMap (void)
 	map_physicsnormals = NULL;
 }
 
+static vfsfile_t *CM_OpenMap(char *name, dheader_t *header)
+{
+#ifndef CLIENTONLY
+	extern cvar_t sv_bspversion, sv_halflifebsp;
+#endif
+	vfsfile_t *vf;
+	int i;
+
+	vf = FS_OpenFile(name);
+	if (!vf) {
+		Host_Error ("CM_OpenMap: %s not found", name);
+	}
+
+	VFS_READ(vf, header, sizeof(dheader_t), NULL);
```

Comment:
Since we doesn’t check results of VFS_READ it would be sane to memset header to zero, or check results of the read.

Unrelated to refactoring you did though, rather fixing original code.

Subsequent commits contains same issue with not checking read/seek calls results but yet again it is rather issue with original code and out of the scope of this change.

Reply:
I think this is one read that should have error handling, which as you mentioned didn't have error handling before. Good catch. Adding a commit to fix this, together with zeroing out the header. Will make the loading more robust than before this PR.

As for the rest of the seek/read, yeah, was leaning back and forth on that based on behavior in rest of the application. Many parts assume the best, so went with that to reduce the diff. I could add improved handling for that in a separate PR, it's not worse than before as you say.

Reply:
bspx header (later commits) probably requires the same additional check and memzero, I guess